### PR TITLE
New employers form

### PR DESCRIPTION
**Pull request by db0sch**
Replace Typeform with a in-house built form on the `/employers` page.
Design is based on Apply form.
Form is encapsulated in a modal.
No real model : Struct with validation that acts like a ActiveRecord model (but it's not - useful cause you can use simple_form)
Once submitted: sends a notification in `#hiring` channel on Le Wagon Team slack
If it doesn't pass the validation, it return the form with errors messages.

some questions for you @ssaunier :
1. I wanted to add some js script to re-show the modal if the controller returns the form due to failed validations.
But, because of `async: true` on the `javascript_tag` on the layout, the js is loading late. And I can't access neither my js function on jquery when the page load.
If you have any tips on how to deal with that, i'm all ears

2. i used the bootstrap 3 convention. but i think we are migrating from 3 to 4 now (but maybe only on Kitt)

3. I'm still getting the city with this instance variable set in the initializer and application_controller (`@cities`). Should I change that, due to you fairly recent alumni migration 😝  ?

form

### CODE REVIEW

**Comment by db0sch** [comment]
I rebased. And nothing works. Will open this PR again when I fixed everything

**Comment by db0sch** [comment]
Fixed the problem. Let's see if Travis will appreciate 🤔

**Comment by db0sch** [comment]
Fixes #522

File: spec/controllers/employers_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      email: "alan@turing.com",
+      company: "E Corp.",
+      website: "www.ecorp.com",
+      targets: ["FullStack Developer", "Product Manager"],
+      locations: ["Tokyo", "Paris"],
+      message: "Realllllyyyyyy long text of more"
+    }
+    employer = Employer.from_hash(params)
+    post :create, params: { employer: params }
+    # expect(NotifySlack).to have_enqueued_sidekiq_job(channel: "hiring", text: employer.to_slack_message)
```
Comment (ssaunier) [inline spec/controllers/employers_controller_spec.rb]: Do you want to keep that code?

File: spec/features/employers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
```
Comment (ssaunier) [inline spec/features/employers_spec.rb]: :+1:

File: app/models/employer.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Employer < Struct.new(:first_name, :last_name, :email, :phone_number, :company, :website, :targets, :locations, :message)
```
Comment (ssaunier) [inline app/models/employer.rb]: You can do a real model. I'd name it `EmployerProspect`, so that we keep the history of people submitting this form.

Comment (db0sch) [reply app/models/employer.rb]: So we should save them to the DB. 🙆‍♂️ 
I wasn't sure about that.
No prob. I'll do that next week :v:

File: spec/controllers/employers_controller_spec.rb (hunk 1 above)
Comment (db0sch) [reply spec/controllers/employers_controller_spec.rb]: nope. it just didn't work for some reason. I'll delete that in my next PR.

**Comment by db0sch** [comment]
@ssaunier I think i'm done.
I serialize the arrays of jobs skills and cities as we don't have them in the `www` database (and therefore can't do an association). Hope this is ok. Let me know.
Thanks 🙏

**Comment by db0sch** [comment]
FYI, the form is now submitted in AJAX.
It allows me to render the form with error while the modal is still open (without AJAX, the page was reloaded and the modal was close. not a very good ui).

Two gifs to show you how it behaves:

With errors:
error

Submit with success:
thank

File: app/models/employer_prospect.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+#
+
+class EmployerProspect < ApplicationRecord
+
+  validates :first_name, :last_name, :phone_number, :company, :website, :targets, :locations, :message, presence: true
+  validates :email, format: /\A([\w+\-].?)+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+  validates :website, format: /\A(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?\z/
+
+  # targets and locations are arrays of strings
+  serialize :targets
```
Comment (ssaunier) [inline app/models/employer_prospect.rb]: You can drop the 2 serializes and have `array: true` migration.

**Comment by ssaunier** [comment]
👏

**Pull request merged**